lib/adapters/REST/endpoints/upload.ts

```diff
@@ -25,18 +28,22 @@ export const create: RestEndpoint<'Upload', 'create'> = (
 
 export const del: RestEndpoint<'Upload', 'delete'> = (
   http: AxiosInstance,
-  params: GetSpaceParams & { uploadId: string }
+  params: GetSpaceEnvironmentParams & { uploadId: string }
 ) => {
   const httpUpload = getUploadHttpClient(http)
-
-  return raw.del(httpUpload, `/spaces/${params.spaceId}/uploads/${params.uploadId}`)
+  const spacePath = `/spaces/${params.spaceId}/uploads/${params.uploadId}`
+  const environmentPath = `/spaces/${params.spaceId}/environments/${params.environmentId}/uploads/${params.uploadId}`
+  const path = params.environmentId ? environmentPath : spacePath
+  return raw.del(httpUpload, path)
 }
 
 export const get: RestEndpoint<'Upload', 'get'> = (
   http: AxiosInstance,
-  params: GetSpaceParams & { uploadId: string }
+  params: GetSpaceEnvironmentParams & { uploadId: string }
 ) => {
   const httpUpload = getUploadHttpClient(http)
-
-  return raw.get(httpUpload, `/spaces/${params.spaceId}/uploads/${params.uploadId}`)
+  const spacePath = `/spaces/${params.spaceId}/uploads/${params.uploadId}`
```

Comment:
This is the third repetition of the same code and I think it would not be premature to abstract it into a helper function (eg in this file)